### ISSUE TITLE
loss terms

### ISSUE THREAD

**Issue by ShengyuH** [chrischoy/FCGF#40]
hi Chris,

In the contrastive loss implementation, for positive part, you use F.relu(squared_distance - positive_margin); for negative part, you use F.relu(negative_margin - squared_distance)**2, is this intentional? 

https://github.com/chrischoy/FCGF/blob/1a4e35c8e11ccdd3fcc9cbe4aaed7a492b5584c5/lib/trainer.py#L447

best,
Shengyu

**Comment by chrischoy** [comment]
Yes, this was intentional since applying square root to get distance is sometimes unstable.
